### ISSUE TITLE
[Android] The Kotlin Gradle plugin was loaded multiple times in different subprojects

### ISSUE THREAD

**Issue by bulkinav** [software-mansion/react-native-gesture-handler#2307]
### Description

Every time when I building my project, the following warning appears in the gradle output. I don't think this a problem (the project build correctly), but decided to inform about it.

My environment:

gradle 7.5.1
gradle plugin: 7.2.2

### Steps to reproduce

Start build the RN project -> the following warning appears in the gradle output:

> The Kotlin Gradle plugin was loaded multiple times in different subprojects, which is not supported and may break the build. 
> This might happen in subprojects that apply the Kotlin plugins with the Gradle 'plugins { ... }' DSL if they specify explicit versions, even if the versions are equal.
> Please add the Kotlin plugin to the common parent project or the root project, then remove the versions in the subprojects.
> If the parent project does not need the plugin, add 'apply false' to the plugin line.
> See: https://docs.gradle.org/current/userguide/plugins.html#sec:subprojects_plugins_dsl
> The Kotlin plugin was loaded in the following projects: ':react-native-gesture-handler', ':react-native-screens'

### Snack or a link to a repository

none

### Gesture Handler version

2.8.0

### React Native version

0.70.4

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

React Native (without Expo)

### Architecture

Paper (Old Architecture)

### Build type

Release mode

### Device

Real device

### Acknowledgements

Yes

**Comment by NiharR27** [comment]
+1 seeing the same issue

**Comment by SaraChicaD** [comment]
might be related to this: https://github.com/facebook/react-native/issues/35210

**Comment by bulkinav** [comment]
@SaraChicaD no, this issue is not related to something like that. The gradle log shows modules that are involved in the issue:

> The Kotlin plugin was loaded in the following projects: ':react-native-gesture-handler', ':react-native-screens'

**Comment by vishaletm** [comment]
+1 Same issue

**Comment by gomezmark** [comment]
any updates?

**Comment by sweatherall** [comment]
+1 also getting this same error:
**_The Kotlin plugin was loaded in the following projects: ':react-native-gesture-handler', ':rnmapbox_maps'_**

**Comment by sweatherall** [comment]
FWIW I have (temporarily?) resolved this issue by adding `kotlinVersion=1.8.0` to my `gradle.properties` file 
(I was having version issues between `jetified-kotlin-stdlib-1.8.0 (org.jetbrains.kotlin:kotlin-stdlib:1.8.0)` and `jetified-kotlin-stdlib-jdk8-1.6.21 (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.6.21)`)

**Comment by sweatherall** [comment]
Update: 
I tested the fix suggested [here](https://kotlinlang.org/docs/whatsnew18.html#usage-of-the-latest-kotlin-stdlib-version-in-transitive-dependencies).
Which is to add `implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.8.0"))` to the `app/build.gradle` file.
This also results in a successful build..

**Comment by zahmadz** [comment]
in my case, i forgot to put Gradle variables into android/gradle.properties.

so just put the Gradle variables into android/gradle.properties.

this is the Gradle variables look like: https://reactnative.dev/docs/signed-apk-android#setting-up-gradle-variables

MYAPP_UPLOAD_STORE_FILE=my-upload-key.keystore
MYAPP_UPLOAD_KEY_ALIAS=my-key-alias
MYAPP_UPLOAD_STORE_PASSWORD=*****
MYAPP_UPLOAD_KEY_PASSWORD=*****

**Comment by kockok** [comment]
Adding `implementation(platform("org.jetbrains.kotlin:kotlin-bom:1.8.0"))` to the app/build.gradle file not working for me. Any updates?

**Comment by recepaltas** [comment]
Any Solution ??

**Comment by AbdulMoizJawed** [comment]
+1 Facing same problem

**Comment by elencho** [comment]
+1

**Comment by Gijuno** [comment]
Same here with **reanimated**

**Comment by vitorpereiira** [comment]
+1

**Comment by sadewole** [comment]
facing this issue too

**Comment by vincentRouget** [comment]
Idem +1

**Comment by kockok** [comment]
I set version to `2.9.1` and then the app successfully built, however, it kept crashing... iOS no such problem.

**Comment by luk3skyw4lker** [comment]
I have the same problem, is there any update on this?

**Comment by tugceaktepe** [comment]
+1

**Comment by DevKalebe** [comment]
Estou com o mesmo problema ainda ao tentar executar meu aplicativo

**Comment by Steentoons** [comment]
+1 Also facing the same problem here, with reanimated

**Comment by dnelsonlivecurrence** [comment]
I believe this could be resolved by making the following modifications to the build.gradle file:

```
buildscript {
  // Define Kotlin Version from parent project - Buildscript is evaluated before everything else so we can't use getExtOrDefault
  def kotlin_version = rootProject.ext.has("kotlinVersion") ? rootProject.ext.get("kotlinVersion") : project.properties["Misnap_kotlinVersion"]

  repositories {
    google()
    mavenCentral()
  }
  dependencies {
    classpath "com.android.tools.build:gradle:7.2.1"
    // noinspection DifferentKotlinGradleVersion
    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
  }
}
```

And then for dependencies:

```
def kotlin_version = getExtOrDefault("kotlinVersion")

dependencies {
  implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
```

**Comment by irmakcosarsahna** [comment]
I am also facing the same problem, but we are able to run it on a different computer. We have checked various settings on Android Studio's side, and everything appears to be the same.

**Comment by soroysoroy4** [comment]
any updates?  does anyone here have a solution?

**Comment by irmakcosarsahna** [comment]
**Solution Update:** 

I'm pleased to share that I've managed to resolve this issue by adding the following line to the `dependencies` section within the `app/build.gradle` file:

```gradle
classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.0")
```

This addition successfully addressed the problem I was encountering.

**Comment by ntalamdotcom** [comment]
I just created a new project using expo, and run it.
I have touched nothing:

`The Kotlin Gradle plugin was loaded multiple times in different subprojects, which is not supported and may break the build.
This might happen in subprojects that apply the Kotlin plugins with the Gradle 'plugins { ... 
}' DSL if they specify explicit versions, even if the versions are equal.
Please add the Kotlin plugin to the common parent project or the root project, then remove the versions in the subprojects.
If the parent project does not need the plugin, add 'apply false' to the plugin line.
See: https://docs.gradle.org/current/userguide/plugins.html#sec:subprojects_plugins_dsl       
The Kotlin plugin was loaded in the following projects: ':expo', ':expo-modules-core'`

**Comment by victorbamikole** [comment]
I'm having the same error but this doesn't seem to solve my issue...

**Comment by JASH-PATEL-6566** [comment]
+1

**Comment by moedeveloper** [comment]
any update same issue!

**Comment by alexxp308** [comment]
I added this line in **android/build.gradle** and it worked

`buildscript {`

    ext {
        ...
        kotlinVersion = findProperty('android.kotlinVersion') ?: '1.8.10'
    }
    ...
    dependencies {
        ...
        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")  //<--------------------------
    }
`}`
I am using react native version 0.72.5

**Comment by maykelnekel** [comment]
Downgrade Java sdk from 21 to 17 work to me on win11

**Comment by w4t3r-45** [comment]
same issue

**Comment by PratikGondaliya007** [comment]
**Error**: The Kotlin Gradle plugin was loaded multiple times in different

1) Open your Project level **build.gradle** file.
2) Inside buildScript{} block write 
`ext{
 kotlinVersion = findProperty('android.kotlinVersion') ?: '1.9.10'
}`
and Inside **dependencies** section paste 
  `classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion"`
3) This way you will resolve error

NOTE: Kotin version change as per your requirements.

I hope it works for you!!!!!

**Comment by ahirner** [comment]
What's the solution for a `build.gradle.kts` where `kotlin = "2.1.20"` is already in `gradle/libs.versions.toml`?

**Comment by signor-mike** [comment]
Not with react-native, but with Capacitor.

I faced similar issue and it got resolved after I updated my `$JAVA_HOME` to match version 21. 
`sudo nano ~/.bashrc` and change -17- (older version) to -21- in my case.
As well as run `sudo update-alternatives --config java` and the same for `javac` to have once again version 21.

I hope this helps somebody.

**Comment by m-bert** [comment]
Hi! I know it's been a while, but it would be great if you could confirm that #4248 fixes this issue.
